FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/DemoTasks/FleetProvisioningDemoExample.c

```diff
@@ -0,0 +1,831 @@
+                 NULL );                   /* Used to pass out a handle to the created task - not used in this case. */
+}
+
+/* This example uses a single application task, which shows that how to use
+ * the Fleet Provisioning library to generate and validate AWS IoT Fleet
+ * Provisioning MQTT topics, and use the coreMQTT library to communicate with
+ * the AWS IoT Fleet Provisioning APIs. */
+int prvFleetProvisioningTask( void * pvParameters )
+{
+    bool xStatus = false;
+    /* Buffer for holding the CSR. */
+    char pcCsr[ fpdemoCSR_BUFFER_LENGTH ] = { 0 };
+    size_t xCsrLength = 0;
+    /* Buffer for holding received certificate until it is saved. */
+    char pcCertificate[ fpdemoCERT_BUFFER_LENGTH ];
+    size_t xCertificateLength;
+    /* Buffer for holding the certificate ID. */
+    char pcCertificateId[ fpdemoCERT_ID_BUFFER_LENGTH ];
+    size_t xCertificateIdLength;
+    /* Buffer for holding the certificate ownership token. */
+    char pcOwnershipToken[ fpdemoOWNERSHIP_TOKEN_BUFFER_LENGTH ];
+    size_t xOwnershipTokenLength;
+    bool xConnectionEstablished = false;
+    CK_SESSION_HANDLE xP11Session;
+    uint32_t ulDemoRunCount = 0U;
+    CK_RV xPkcs11Ret = CKR_OK;
+
+
+    NetworkContext_t xNetworkContext = { 0 };
+    TlsTransportParams_t xTlsTransportParams = { 0 };
+
+    /* Silence compiler warnings about unused variables. */
+    ( void ) pvParameters;
+
+    /* Set the pParams member of the network context with desired transport. */
+    xNetworkContext.pxParams = &xTlsTransportParams;
+
+    do
+    {
+        /* Initialize the buffer lengths to their max lengths. */
+        xCertificateLength = fpdemoCERT_BUFFER_LENGTH;
+        xCertificateIdLength = fpdemoCERT_ID_BUFFER_LENGTH;
+        xOwnershipTokenLength = fpdemoOWNERSHIP_TOKEN_BUFFER_LENGTH;
+
+        /* Initialize the PKCS #11 module */
+        xPkcs11Ret = xInitializePkcs11Session( &xP11Session );
+
+        if( xPkcs11Ret != CKR_OK )
+        {
+            LogError( ( "Failed to initialize PKCS #11." ) );
+            xStatus = false;
+        }
+        else
+        {
+            xStatus = xGenerateKeyAndCsr( xP11Session,
+                                          pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+                                          pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS,
+                                          pcCsr,
+                                          fpdemoCSR_BUFFER_LENGTH,
+                                          &xCsrLength );
+
+            if( xStatus == false )
+            {
+                LogError( ( "Failed to generate Key and Certificate Signing Request." ) );
+            }
+
+            xPkcs11CloseSession( xP11Session );
+        }
+
+        /**** Connect to AWS IoT Core with provisioning claim credentials *****/
+
+        /* We first use the claim credentials to connect to the broker. These
+         * credentials should allow use of the RegisterThing API and one of the
```

Comment:
Do we provide an example of generating the claim certificate / private key and assigning the correct policy to that claim certificate?

Reply:
Yes - the documentation will include examples and links for this information.